numpy/lib/tests/test_format.py

```diff
@@ -428,6 +428,17 @@ def roundtrip_truncated(arr):
     arr2 = format.read_array(f2)
     return arr2
 
+def file_truncated(arr):
+    with temppath() as path:
+        with open(path, 'wb') as f:
+            format.write_array(f, arr)
+        #truncate the file by one byte
+        with open(path, 'rb+') as f:
+            f.seek(-1, os.SEEK_END)
+            f.truncate()
+        with open(path, 'rb') as f:
+            arr2 = format.read_array(f)
+        return arr2
```

Comment:
Please move this into the test.  Also use the `tmppath` fixture (i.e. a "magic" argument that pytest automatically passes if it finds it).